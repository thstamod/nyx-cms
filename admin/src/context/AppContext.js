import React, { createContext, useContext } from 'react';
import { userReducer, initialState } from '../redux/reducers/userReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const AppContext = createContext();

export function useAppState() {
  return useContext(AppContext);
}

export function AppStateProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(userReducer, initialState, middlewares);

  return <AppContext.Provider value={store}>{children}</AppContext.Provider>;
}
