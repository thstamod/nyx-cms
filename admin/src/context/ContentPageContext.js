import React, { createContext, useContext } from 'react';
import {
  initialState,
  contentPageReducer,
} from '../state/reducers/contentPageReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const ContentPageContext = createContext(null);

export function useContentPageState() {
  return useContext(ContentPageContext);
}

export function ContentPageProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(
    contentPageReducer,
    initialState,
    middlewares
  );

  return (
    <ContentPageContext.Provider value={store}>
      {children}
    </ContentPageContext.Provider>
  );
}
