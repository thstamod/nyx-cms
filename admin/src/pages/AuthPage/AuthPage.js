/* eslint-disable react/destructuring-assignment */
import React, { useState } from 'react';
// import { useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { loginAction } from '../../redux/actions/userActions';
import { calculateExpirationTime } from '../../utils/calculateTime';
import withData from '../../containers/withData';
import LOGIN_QUERY from '../../graphql/loginQuery';
import { useAppState } from '../../context/AppContext';

const AuthPage = (props) => {
  const [, dispatch] = useAppState();
  // const dispatch = useDispatch();
  const [email, setemail] = useState('');
  const [password, setpassword] = useState('');

  const handleError = (err) =>
    err && (
      <div>
        Bad:
        {err.graphQLErrors.map(({ message }, i) => (
          <span key={i}>{message}</span>
        ))}
      </div>
    );

  if (props.data && props.data.login) {
    dispatch(
      loginAction(
        props.data.login.token,
        calculateExpirationTime(props.data.login.tokenExpiration)
      )
    );
    return <Redirect to="/content" />;
  }
  return (
    <form>
      <div>
        <label htmlFor="email">Email</label>
        <input
          type="email"
          id="email"
          onChange={(e) => {
            setemail(e.target.value);
          }}
        />
        <label htmlFor="password">Password</label>
        <input
          type="password"
          id="password"
          onChange={(e) => {
            setpassword(e.target.value);
          }}
        />
      </div>
      {handleError(props.error)}
      <button
        type="button"
        onClick={() =>
          props.handleSubmit({
            variables: {
              email,
              password,
            },
          })
        }
      >
        Submit
      </button>
    </form>
  );
};

export default withData({ query: LOGIN_QUERY, lazy: true })(AuthPage);
