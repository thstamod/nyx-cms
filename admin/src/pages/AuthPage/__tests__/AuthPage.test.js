import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { AppStateProvider } from '../../../context/AppContext';

import AuthPage from '../AuthPage';

test('Auth page Page renders', () => {
  const { asFragment } = render(
    <AppStateProvider>
      <AuthPage />
    </AppStateProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});
