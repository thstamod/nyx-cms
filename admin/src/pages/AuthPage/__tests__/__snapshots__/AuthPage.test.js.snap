// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth page Page renders 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        type="email"
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
